robot/impl/resource_manager.go

```diff
@@ -836,6 +836,15 @@ func (manager *resourceManager) processRemote(
 	config config.Remote,
 	gNode *resource.GraphNode,
 ) (*client.RobotClient, error) {
+	// if there was an existing client (i.e. remote was modified), close old client before making a new one
+	res, err := gNode.Resource()
```

Comment:
cc @maximpertsov will this at some point return an error when a remote is "unhealthy"? I assume we should still `Close` "unhealthy" remotes. Maybe we should just use `IsUnitialized` to check if there's an actual client here?

Reply:
Yeah that's a good question - I haven't settled on the new semantics of `GraphNode.Resource` once we add the concept of "unhealthy" nodes, but  I'm leaning towards towards keeping it the same. I think this logic is fine as is, and the unit test should cover us if the `GraphNode.Resource` api changes.